docs/developing.rst

```diff
@@ -114,3 +114,20 @@ performing these checks locally avoids unnecessary build failures.
     # build the documentation
     make --directory=docs clean html
     $BROWSER ./docs/_build/html/index.html
+
+
+Plugins
+-------
+
+Removing plugins
+^^^^^^^^^^^^^^^^
+
+1. Remove the plugin file in ``src/streamlink/plugins/`` and the test file in ``tests/plugins/``
+2. Remove the plugin entry from the documentation in ``docs/plugin_matrix.rst``
+3. Add an entry to ``src/streamlink/plugins/.removed``
+
+Adding plugins
+^^^^^^^^^^^^^^
+
+1. Implement the plugin in ``src/streamlink/plugins/``, similar to already existing plugins. Check the git log for recently added or modified plugins to help you get an overview of what's needed to properly implement a plugin. A complete guide is currently not available.
+2. Add at least tests for the URL regex matching in ``tests/plugins/``. Once again, check other plugin tests from the git log.
```

Comment:
Max line length defined by the editorconfig is 128:
https://github.com/streamlink/streamlink/blob/8db5786a4f8422688d8618685cc30057a2fa423c/.editorconfig#L9

Reply:
Oops, sorry, I had not realised.  I didn't check it before with the docs build either, but this time I have.